arch/arm/compare256_neon.c

```
@@ -22,33 +22,55 @@ Z_FORCEINLINE static uint32_t compare256_neon_static(const uint8_t *src0, const
+        src0 += 32, src1 += 32;
 #endif
 
-        cmp = veorq_u8(a, b);
+        uint8x16_t cmp1 = veorq_u8(a1, b1);
+        uint8x16_t cmp2 = veorq_u8(a2, b2);
 
-        lane = vgetq_lane_u64(vreinterpretq_u64_u8(cmp), 0);
+#if defined(ARCH_ARM) && defined(ARCH_64BIT)
+        /* 2x unrolled UMAXV fast path: check 32 bytes with a single UMAXV */
+        uint8x16_t any_diff = vorrq_u8(cmp1, cmp2);
+        if (LIKELY(vmaxvq_u8(any_diff) == 0)) {
```

Cppcheck (2.21.0)

[style] The function 'vld1q_u16_x4' is never used.

(unusedFunction)